flake.nix

```diff
@@ -7,6 +7,14 @@
 
   outputs = { self, nixpkgs }: {
 
+    nixosModules.default = {...}: {
+       fileSystems."/" = {
+         device = "/dev/sda1";
+         fsType = "ext4";
+       };
+       boot.loader.grub.device = "/dev/sda";
+    };
```

Comment:
Should we use `lib.mkForce` for *all* the needed settings? Otherwise we risk that someone will silently overwrite the settings. I think this makes sense, since we're assuming that -- without these settings -- deployment just breaks.

Reply:
Also, maybe add some tests?

Reply:
What kind of tests do you have in mind? Or are the assertions enough for now?

Reply:
@eviefp if you're taking over, I can push a branch where I had started fixing merge conflicts are writing tests

Reply:
https://github.com/garnix-io/garnix-lib/compare/jkarni/garnix-module2?expand=1

Reply:
> What kind of tests do you have in mind? Or are the assertions enough for now?

I thought maybe just have *some* automated test that a nixos configuration using `garnix.server.enable = true;` actually builds? E.g. we could just add an example configuration to this flake that'll be built on garnix CI.